radio/src/audio_arm.h

```diff
@@ -254,14 +254,15 @@ class MixedContext {
       tone.clear();   // the biggest member of the uninon
     }
 
+    bool isEmpty() const { return fragment.type == FRAGMENT_EMPTY; };
     bool isTone() const { return fragment.type == FRAGMENT_TONE; };
     bool isFile() const { return fragment.type == FRAGMENT_FILE; };
     bool hasId(uint8_t id) const { return fragment.id == id; };
 
-    int mixBuffer(AudioBuffer *buffer, int volume, unsigned int fade)
+    int mixBuffer(AudioBuffer *buffer, int toneVolume, int wavVolume, unsigned int fade)
```

Comment:
Bummer, I kind of like the new solution, the code in the wakeup function is much clearer now. Please consider changing your opinion.